scipy/optimize/_bracket.py

```diff
@@ -378,3 +380,277 @@ def customize_result(res, shape):
     return eim._loop(work, callback, shape, maxiter, func, args, dtype,
                      pre_func_eval, post_func_eval, check_termination,
                      post_termination_check, customize_result, res_work_pairs)
+
+
+def _bracket_minimum_iv(func, xm0, xl0, xr0, xmin, xmax, factor, args, maxiter):
+
+    if not callable(func):
+        raise ValueError('`func` must be callable.')
+
+    if not np.iterable(args):
+        args = (args,)
+
+    xm0 = np.asarray(xm0)[()]
+    if not np.issubdtype(xm0.dtype, np.number) or np.iscomplex(xm0).any():
+        raise ValueError('`xm0` must be numeric and real.')
+
+    xmin = -np.inf if xmin is None else xmin
+    xmax = np.inf if xmax is None else xmax
+
+    xl0 = xm0 - 0.5 if xl0 is None else xl0
+    xr0 = xm0 + 0.5 if xr0 is None else xr0
+
+    factor = 2.0 if factor is None else factor
+    xl0, xm0, xr0, xmin, xmax, factor = np.broadcast_arrays(
+        xl0, xm0, xr0, xmin, xmax, factor
+    )
+
+    if not np.issubdtype(xl0.dtype, np.number) or np.iscomplex(xl0).any():
+        raise ValueError('`xl0` must be numeric and real.')
+
+    if not np.issubdtype(xr0.dtype, np.number) or np.iscomplex(xr0).any():
+        raise ValueError('`xr0` must be numeric and real.')
+
+    if not np.issubdtype(xmin.dtype, np.number) or np.iscomplex(xmin).any():
+        raise ValueError('`xmin` must be numeric and real.')
+
+    if not np.issubdtype(xmax.dtype, np.number) or np.iscomplex(xmax).any():
+        raise ValueError('`xmax` must be numeric and real.')
+
+    if not np.issubdtype(factor.dtype, np.number) or np.iscomplex(factor).any():
+        raise ValueError('`factor` must be numeric and real.')
+    if not np.all(factor > 1):
+        raise ValueError('All elements of `factor` must be greater than 1.')
+
+    # Default choices for xl or xr might have exceeded xmin or xmax. Adjust
+    # to make sure this doesn't happen. We replace with copies because xl, and xr
+    # are read-only views produced by broadcast_arrays.
+    xl0, xr0 = xl0.copy(), xr0.copy()
+    cond = ~np.isinf(xmin) & (xl0 < xmin)
+    xl0[cond] = (
+        xm0[cond] - xmin[cond]
+    ) / np.array(16, dtype=xl0.dtype)
+    cond = ~np.isinf(xmax) & (xmax < xr0)
+    xr0[cond] = (
+        xmax[cond] - xm0[cond]
+    ) / np.array(16, dtype=xr0.dtype)
+
+    maxiter = np.asarray(maxiter)
+    message = '`maxiter` must be a non-negative integer.'
+    if (not np.issubdtype(maxiter.dtype, np.number) or maxiter.shape != tuple()
+            or np.iscomplex(maxiter)):
+        raise ValueError(message)
+    maxiter_int = int(maxiter[()])
+    if not maxiter == maxiter_int or maxiter < 0:
+        raise ValueError(message)
+
+    if not np.all((xmin <= xl0) & (xl0 < xm0) & (xm0 < xr0) & (xr0 <= xmax)):
+        raise ValueError(
+            '`xmin <= xl0 < xm0 < xr0 <= xmax` must be True (elementwise).'
+        )
```

Comment:
This is fine for now, but I think I did this at some point when I was feeling lazy and set a bad precedent. Bracket calculation should continue but fail for these slices.

Reply:
Yeah, that does sound better.